### DISCUSSION TITLE
API documentation - Question

### DISCUSSION THREAD

**Discussion by sailujanu** [go-kit/kit#1137] [Q&A]
Hi, 

We are currently using go-kit in our project and it is great so far. We wanted to generate API documentation and I have gone through goswagger https://github.com/go-swagger/go-swagger and according to this it need to add annotations in order to generate spec but to add annotation it should be at controller level. go-kit follows gPRC and not sure how to add annotations. Does anyone know how to generate API documentation for go-kit and if any sample projects or reference would be helpful. 

Thanks in Advance. !!

**Comment by alihalabyah** [comment]
Considering this as an old question, Does the most recent Go Kit support swagger documentation or not yet?

**Comment by peterbourgon** [comment]
Despite my repeated attempts to find and leverage any overlap, there is really no connection between Swagger (OpenAPI) and Go kit.

**Comment by SantoshSah** [comment]
@peterbourgon , does go-kit support any other API Documentation Softwares?

**Comment by SantoshSah** [comment]
Any update on this issue?

**Comment by nelz9999** [comment]
@SantoshSah Peter said no to OpenAPI above. What else are you interested in? Do you have examples? 

Maybe if you provided some more context around what you are trying to achieve, someone could give you a more apt answer.

**Comment by SantoshSah** [comment]
@nelz9999 , i am trying to achieve automated API documentation for go-kit services.

**Comment by eldad87** [comment]
It will be greater if we can define the service (gRPC & REST) using protobuf and generate our swaggers using it.
Similar to:
https://github.com/grpc-ecosystem/grpc-gateway
https://github.com/eldad87/go-boilerplate
